### DISCUSSION TITLE
Audiobookshelf Integration

### DISCUSSION THREAD

**Discussion by SAS-1** [IgnisDa/ryot#529] [Q&A] [answered]
Hi

I just found out about ryot and want to try it out so I have created my compose file and got it started but I can't get ABS to work. 

I have set the integration to use both internal IP:port and the external url to ABS with my token but when I go to Audio Books and Podcasts nothing is showing. 

I have clicked on Synchronize integrations progress as well but nothing is importing.

I can't find any logs to check what is happening any ideas?

Thanks

**Comment by IgnisDa** [comment]
Is ABS accessible from inside the container?

**Comment by SAS-1** [comment]
yeah they are on the same Docker Network and I can run curl audiobookshelf and it returns the correct page (ping doesn't seem to be in the docker image so can't use that)

I stuck this on the environment variables
      - RUST_LOG=ryot=trace,sea_orm=debug

but I can't find the log file in docker or Loki - I send all my logs to Loki using

   logging:
      driver: loki
      options:
       loki-url: "http://LOKI/loki/api/v1/push"

**Reply by IgnisDa** [reply]
> I can run curl audiobookshelf 

Where are you running curl? The `v3.*` ryot images do not have any shell. Are you accessing it from the host machine/VPN?

**Reply by IgnisDa** [reply]
Also, I suggest updating Ryot to v4 which was just released.

**Comment by SAS-1** [comment]
> > I can run curl audiobookshelf
> 
> Where are you running curl? The `v3.*` ryot images do not have any shell. Are you accessing it from the host machine/VPN?

I stuck V4 on this morning and used the shell via Portainer

**Reply by IgnisDa** [reply]
Ahh okay. 

Can you run `curl http://container-url:8000/backend/config` from audiobookshelf and show me the output?
And also curl audiobookshelf from ryot.

**Comment by SAS-1** [comment]
yep from ABS I get 

curl http://IP:8002/backend/config

{"anime_and_manga":{"anilist":{"prefer_english":false},"mal":{"client_id":"****"},"manga_updates":{}},"audio_books":{"audible":{"locale":"us"}},"books":{"openlibrary":{"cover_image_size":"M"},"google_books":{}},"database":{"url":"****"},"exercise":{},"file_storage":{"s3_access_key_id":"****","s3_bucket_name":"****","s3_region":"****","s3_secret_access_key":"****","s3_url":"****"},"frontend":{"url":"https://ryot.fly.dev","item_details_height":300,"page_size":20,"umami":{"script_url":"","website_id":"","domains":""}},"integration":{"pull_every":2,"hasher_salt":"****","minimum_progress_limit":2,"maximum_progress_limit":95},"media":{},"movies_and_shows":{"tmdb":{"access_token":"****","locale":"en"}},"podcasts":{"listennotes":{"api_token":"****"},"itunes":{"locale":"en_us"}},"scheduler":{"database_url":"****","rate_limit_num":5,"user_cleanup_every":12},"server":{"config_dump_path":"****","cors_origins":["****"],"default_credentials":false,"progress_update_threshold":2,"person_outdated_threshold":30,"deploy_admin_jobs_allowed":true,"max_file_size":70,"videos_disabled":false,"update_monitored_media":true,"graphql_playground_enabled":true},"users":{"jwt_secret":"****","allow_changing_preferences":true,"allow_changing_credentials":true,"allow_registration":true,"reviews_disabled":false,"token_valid_for_days":90},"video_games":{"igdb":{"image_size":"t_original"},"twitch":{"client_id":"****","client_secret":"****"}},"visual_novels":{}}/

from Ryot if I use curl audiobookshelf

I get

curl audiobookshelf
<!doctype html>
<html lang="en" data-n-head="%7B%22lang%22:%7B%221%22:%22en%22%7D%7D">
  <head>
    <title>Audiobookshelf</title><meta data-n-head="1" charset="utf-8"><meta data-n-head="1" name="viewport" content="width=device-width,initial-scale=1"><meta data-n-head="1" data-hid="description" name="description" content=""><meta data-n-head="1" data-hid="charset" charset="utf-8"><meta data-n-head="1" data-hid="mobile-web-app-capable" name="mobile-web-app-capable" content="yes"><meta data-n-head="1" data-hid="apple-mobile-web-app-capable" name="apple-mobile-web-app-capable" content="yes"><meta data-n-head="1" data-hid="apple-mobile-web-app-status-bar-style" name="apple-mobile-web-app-status-bar-style" content="black"><meta data-n-head="1" data-hid="apple-mobile-web-app-title" name="apple-mobile-web-app-title" content="Audiobookshelf"><meta data-n-head="1" data-hid="theme-color" name="theme-color" content="#373838"><meta data-n-head="1" data-hid="og:type" name="og:type" property="og:type" content="website"><meta data-n-head="1" data-hid="og:title" name="og:title" property="og:title" content="Audiobookshelf"><meta data-n-head="1" data-hid="og:site_name" name="og:site_name" property="og:site_name" content="Audiobookshelf"><link data-n-head="1" rel="icon" type="image/x-icon" href="/favicon.ico"><link data-n-head="1" data-hid="shortcut-icon" rel="shortcut icon" href="/icon.svg"><link data-n-head="1" data-hid="apple-touch-icon" rel="apple-touch-icon" href="/icon64.png" sizes="64x64"><link data-n-head="1" rel="manifest" href="/_nuxt/manifest.31bfdee8.json" data-hid="manifest"><base href="/"><link rel="preload" href="/_nuxt/144f92b.js" as="script"><link rel="preload" href="/_nuxt/d803ba8.js" as="script"><link rel="preload" href="/_nuxt/5195a29.js" as="script"><link rel="preload" href="/_nuxt/ec8c9a0.js" as="script">
  </head>
  <body>
    <div id="__nuxt"><style>#nuxt-loading{background:#fff;visibility:hidden;opacity:0;position:absolute;left:0;right:0;top:0;bottom:0;display:flex;justify-content:center;align-items:center;flex-direction:column;animation:nuxtLoadingIn 10s ease;-webkit-animation:nuxtLoadingIn 10s ease;animation-fill-mode:forwards;overflow:hidden}@keyframes nuxtLoadingIn{0%{visibility:hidden;opacity:0}20%{visibility:visible;opacity:0}100%{visibility:visible;opacity:1}}@-webkit-keyframes nuxtLoadingIn{0%{visibility:hidden;opacity:0}20%{visibility:visible;opacity:0}100%{visibility:visible;opacity:1}}#nuxt-loading>div,#nuxt-loading>div:after{border-radius:50%;width:5rem;height:5rem}#nuxt-loading>div{font-size:10px;position:relative;text-indent:-9999em;border:.5rem solid #f5f5f5;border-left:.5rem solid #000;-webkit-transform:translateZ(0);-ms-transform:translateZ(0);transform:translateZ(0);-webkit-animation:nuxtLoading 1.1s infinite linear;animation:nuxtLoading 1.1s infinite linear}#nuxt-loading.error>div{border-left:.5rem solid #ff4500;animation-duration:5s}@-webkit-keyframes nuxtLoading{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes nuxtLoading{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}</style><script>window.addEventListener("error",function(){var e=document.getElementById("nuxt-loading");e&&(e.className+=" error")})</script><div id="nuxt-loading" aria-live="polite" role="status"><div>Loading...</div></div></div><script>window.__NUXT__={config:{version:"2.7.0",routerBasePath:"",_app:{basePath:"/",assetsPath:"/_nuxt/",cdnURL:null}}}</script>
  <script src="/_nuxt/144f92b.js"></script><script src="/_nuxt/d803ba8.js"></script><script src="/_nuxt/5195a29.js"></script><script src="/_nuxt/ec8c9a0.js"></script></body>
</html>

I am running ryot on 8002 as I have something else on 8000

**Reply by IgnisDa** [reply]
Welp. This is a bug in ryot then. Creating an issue.

**Reply by IgnisDa** [reply]
@SAS-1 I see that you're running `curl audiobookshelf`. What exactly are you putting in the form when creating the ABS integration ?

**Comment by SAS-1** [comment]
tried a few things

tried audiobookshelf - but this won't work it gives a 400 bad request
tried http://audiobookshelf:13378 
tried http://IP:13378
tried https://audiobookshelf.mydomain.com

none worked

**Reply by IgnisDa** [reply]
Can you curl the last one from Ryot container and attach the output screenshot?

**Reply by IgnisDa** [reply, marked as answer]
Also please reply on #533 incase someone else faces the same problem.

**Comment by SAS-1** [comment]
reply sent to #533
